### DISCUSSION TITLE
Error creating (importing) app

### DISCUSSION THREAD

**Discussion by oitkenmen** [Budibase/budibase#5556] [Help] [answered]
Hello, I have a problem importing an app from cloud to self-hosted on k8s. I've noticed that budibase-couchdb doesn't have Cluster IP, is this OK? The docs doesn't say about it.

**Comment by oitkenmen** [comment]
Tried deleting the cluster and reinstalling many times. Still no Cluster IP and cannot import the app. What might be the problem? Could anyone please provide a short answer.

**Reply by melohagan** [reply]
Hi @berkimbayev What error messages are you seeing? Would you be able to provide any errors from the Console/Network Tab? 

In addition to reinstalling, I could also wipe the volumes used by your k8 instance to make sure it's completely clean if you can do that.

**Reply by oitkenmen** [reply]
Hello, thanks for an answer. Yes, I've deleted the volumes too and waited until the instances were deleted too. The error is 413 Request Entity Too Large. There is no such error in the cloud version.

This problem appeared after AWS wiped out the account because of the verification issue (for no reason). When the account was recovered, I tried to reinstall the app and import the saved work, but it's not working correctly. As far as I remember, the budibase-couchdb service should have Cluster IP. But it doesn't appear after installation.

**Reply by melohagan** [reply]
What is the file size of your export? There's a max import size of 10MB on all server requests, which includes app importing.

**Reply by oitkenmen** [reply]
3MB

**Comment by shogunpurple** [comment, marked as answer]
Hi @berkimbayev - could you provide some information about:
- Where your K8S Cluster is hosted
- Which version of the budibase helm chart you are using
- Which Kubernetes version you are running
- Your kubernetes configuration and values.yaml (if possible)

Which verification issue are you seeing? Could you provide some more detail on that?

Thanks!

**Reply by oitkenmen** [reply]
Hello, my cluster is hosted on AWS EKS. I use kubectl 1.22 and kubernetes cluster is 1.21 . Could you advise how can I find the helm chart's version? Because I don't know how can I see it. I've just installed it as described in the docs. The similar answer is for kubernetes configuration and values. yaml . I don't know where to see them. I've just installed everything according to the docs. I'm not an advanced user of the terminal, just wanted to use the budibase itself, since it was very intuitive. Haven't configured anything.

**Reply by oitkenmen** [reply]
When I install everything, I just enter `kubectl get pods -n budibase` and `kubectl get svc -n budibase` to verify and copy the URL of the app. Then, I enter the URL, sign up and try to import the app. It shows an error.

When I import the same file in cloud version, it shows no errors. Also, `kubectl get svc -n budibase` shows that budibase-couchdb doesn't have Cluster IP. I've reinstalled many times (even on Windows), but nothing changed.

**Reply by shogunpurple** [reply]
Thanks @berkimbayev - if you followed the instructions you should be on the latest - 2.8.0. You can confirm by using:

`helm list -n budibase`

I just tested this and I'm able to recreate it on one of our kubernetes environments - which is odd because the NGINX config for both environments is exactly the same! 

I did some digging and realised that it wasn't actually our own `proxy-service` that was the problem. The standard budibase helm setup ships with the NGINX ingress controller, which is another NGINX that sits in front of budibase and exposes the kubernetes cluster to the network.

I added the following configuration to the NGINX ingress controller `values.yaml` and re-initalised my ingress controller:
```
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/client-max-body-size: 150M
    nginx.ingress.kubernetes.io/proxy-body-size: 50m
```

This worked - I've pushed the fix and raised a PR (https://github.com/Budibase/budibase/pull/5568) and I will get a self-hosted release out once merged, with the latest helm chart. You should then be able to pull this and import your app successfully.

If you cannot wait until then, you can update your ingress definition manually in kubernetes and run again.
`kubectl edit ingress <name-of-your-ingress> -n budibase`

- Add the annotations above to your ingress controller definition
- save and exit
- reboot your NGINX ingress controller: `kubectl rollout restart deployment <nginx-ingress-controller-deployment> -n budibase`
- It should then work.

Hope this helps!

**Reply by oitkenmen** [reply]
Hello, thanks for the detailed answer, partly I didn't understand it but will reinstall the cluster and see the change 🙏🏻

**Reply by oitkenmen** [reply]
Thank you so much, it finally worked!
